test/Test.hs

```diff
@@ -23,11 +22,19 @@ main = do
 
     if result
         then putStrLn "Success"
-        else putStrLn "Fail" >> exitFailure
+        else putStrLn "Some tests failed" >> exitFailure
   where
     compareWithGolden :: FilePath -> IO Bool
     compareWithGolden f = do
         let (name, _) = splitExtension f
         goldenContent <- readFile $ name <.> "golden"
         testContent   <- readFile f
-        pure $ goldenContent == testContent
+        if goldenContent == testContent
+          then return True
```

Comment:
Just a minor stylistic preference, but you can use `pure` instead of `return` (pure is shorter)